code_examples/sdk_examples/src/workshop/attester/generateDid.ts

```diff
@@ -12,40 +9,33 @@ export async function createFullDid(
   fullDid: Kilt.DidDocument
 }> {
   const api = Kilt.ConfigService.get('api')
-
-  const mnemonic = Kilt.Utils.Crypto.mnemonicGenerate()
-  const {
-    authentication,
-    keyAgreement,
-    assertionMethod,
-    capabilityDelegation
-  } = generateKeypairs(mnemonic)
-  // Get tx that will create the DID on chain and DID-URI that can be used to resolve the DID Document.
-  const fullDidCreationTx = await Kilt.Did.getStoreTx(
-    {
-      authentication: [authentication],
-      keyAgreement: [keyAgreement],
-      assertionMethod: [assertionMethod],
-      capabilityDelegation: [capabilityDelegation]
-    },
-    submitterAccount.address,
-    async ({ data }) => ({
-      signature: authentication.sign(data),
-      keyType: authentication.type
-    })
-  )
-
-  await Kilt.Blockchain.signAndSubmitTx(fullDidCreationTx, submitterAccount)
-
-  const didUri = Kilt.Did.getFullDidUriFromKey(authentication)
-  const encodedFullDid = await api.call.did.query(Kilt.Did.toChain(didUri))
-  const { document } = Kilt.Did.linkedInfoFromChain(encodedFullDid)
-
-  if (!document) {
+  const mnemonic = process.env.ATTESTER_ACCOUNT_MNEMONIC as string
+  const { type, publicKey } = submitterAccount
+
+  const txs = [
+    api.tx.did.createFromAccount({ [type as 'ed25519']: publicKey }),
+    api.tx.did.dispatchAs(
+      submitterAccount.address,
+      api.tx.did.setAttestationKey({ [type as 'ed25519']: publicKey })
+    )
+  ]
+
+  console.log('Creating DID from account…')
+  const didDocument = await Kilt.Blockchain.signAndSubmitTx(
+    api.tx.utility.batch(txs),
+    submitterAccount
+  ).then(async () => {
+    const didUri = Kilt.Did.getFullDidUriFromKey(submitterAccount)
+    const encodedFullDid = await api.call.did.query(Kilt.Did.toChain(didUri))
+    const { document } = Kilt.Did.linkedInfoFromChain(encodedFullDid)
+    return document
+  })
```

Comment:
I don't understand the use of a then clause here - we're in an async context and signAndSubmit is already awaited.
If you wanted to simply rename the variable you could do
```
 const { document: didDocument } = Kilt.Did.linkedInfoFromChain(encodedFullDid)
```
What was the goal for which you introduced it?

Reply:
@rflechtner I thought that was "best practice", but if it's not necessary, I have simplified it.

Reply:
in an async context it is almost always more legible if you simply await other async functions, I'd say